dask_planner/src/sql/optimizer.rs

```diff
@@ -7,40 +7,35 @@ use datafusion_optimizer::{
     decorrelate_where_exists::DecorrelateWhereExists,
     decorrelate_where_in::DecorrelateWhereIn,
     eliminate_cross_join::EliminateCrossJoin,
-    // TODO: need to handle EmptyRelation for GPU cases
-    // eliminate_filter::EliminateFilter,
     eliminate_limit::EliminateLimit,
     eliminate_outer_join::EliminateOuterJoin,
     filter_null_join_keys::FilterNullJoinKeys,
     inline_table_scan::InlineTableScan,
-    limit_push_down::LimitPushDown,
     optimizer::{Optimizer, OptimizerRule},
-    projection_push_down::ProjectionPushDown,
     push_down_filter::PushDownFilter,
+    push_down_limit::PushDownLimit,
+    push_down_projection::PushDownProjection,
     rewrite_disjunctive_predicate::RewriteDisjunctivePredicate,
     scalar_subquery_to_join::ScalarSubqueryToJoin,
     simplify_expressions::SimplifyExpressions,
-    subquery_filter_to_join::SubqueryFilterToJoin,
     type_coercion::TypeCoercion,
     unwrap_cast_in_comparison::UnwrapCastInComparison,
-    OptimizerConfig,
+    OptimizerContext,
 };
 use log::trace;
 
 mod filter_columns_post_join;
-use filter_columns_post_join::FilterColumnsPostJoin;
 
 /// Houses the optimization logic for Dask-SQL. This optimization controls the optimizations
 /// and their ordering in regards to their impact on the underlying `LogicalPlan` instance
 pub struct DaskSqlOptimizer {
-    skip_failing_rules: bool,
     optimizer: Optimizer,
 }
 
 impl DaskSqlOptimizer {
     /// Creates a new instance of the DaskSqlOptimizer with all the DataFusion desired
     /// optimizers as well as any custom `OptimizerRule` trait impls that might be desired.
-    pub fn new(skip_failing_rules: bool) -> Self {
+    pub fn new() -> Self {
```

Comment:
Curious why we removed this

Reply:
That configuration value is no longer exposed via datafusion therefore this was "dead_code" and the compiler issuing warnings and `cargo clippy` was failing

Reply:
Ah they might have renamed that one to `skip_failed_rules`. https://docs.rs/datafusion-common/17.0.0/datafusion_common/config/struct.OptimizerOptions.html

We can update in a followup if needed.